### DISCUSSION TITLE
Which backend do you pair with NextJS?

### DISCUSSION THREAD

**Discussion by enricoschaaf** [vercel/next.js#15607] [Help] [answered]
I am very curious which backends you use for your NextJS applications? Because of Vercels limit to 12 API routes and the lack of backend processing/queueing with Vercel, I tried to AWS CDK, while powerfull, the developer experience is pretty bad compaired to NextJS, to I am looking to try out other alternatives.

**Comment by timneutkens** [comment, marked as answer]
> Because of Vercels limit to 12 API routes

Next.js does smart bundling of API routes and SSR pages so you should not run into the 12 function limit when using Next.js.

**Reply by enricoschaaf** [reply]
Wow, didn't know that, that's nice!

**Reply by kandros** [reply]
@timneutkens nice! Could you please point to where this happens in the codebase please?

**Reply by timneutkens** [reply]
https://github.com/vercel/vercel the Next.js builder
